src/effects/backends/effectmanifestparameter.h

```diff
@@ -0,0 +1,289 @@
+            return "LINKED";
+        } else if (type == LinkType::LINKED_LEFT) {
+            return "LINKED_LEFT";
+        } else if (type == LinkType::LINKED_RIGHT) {
+            return "LINKED_RIGHT";
+        } else if (type == LinkType::LINKED_LEFT_RIGHT) {
+            return "LINKED_LEFT_RIGHT";
+        } else {
+            return "NONE";
+        }
+    }
+
+    static LinkType LinkTypeFromString (const QString& string) {
+        if (string == "LINKED") {
+            return LinkType::LINKED;
+        } else if (string == "LINKED_LEFT") {
+            return LinkType::LINKED_LEFT;
+        } else if (string == "LINKED_RIGHT") {
+            return LinkType::LINKED_RIGHT;
+        } else if (string == "LINKED_LEFT_RIGHT") {
+            return LinkType::LINKED_LEFT_RIGHT;
+        } else {
+            return LinkType::NONE;
+        }
+    }
+
+    enum class LinkInversion {
+        NOT_INVERTED = 0,
+        INVERTED = 1
+    };
+
+    EffectManifestParameter()
+            : m_valueScaler(ValueScaler::UNKNOWN),
+              m_semanticHint(SemanticHint::UNKNOWN),
+              m_unitsHint(UnitsHint::UNKNOWN),
+              m_defaultLinkType(LinkType::NONE),
+              m_defaultLinkInversion(LinkInversion::NOT_INVERTED),
+              m_neutralPointOnScale(0.0),
+              m_default(0),
+              m_minimum(0),
+              m_maximum(1.0) {
+    }
+
+    ~EffectManifestParameter() {
+        //qDebug() << debugString() << "destroyed";
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // Parameter Information
+    ////////////////////////////////////////////////////////////////////////////////
+
+    const QString& id() const {
+        return m_id;
+    }
+    void setId(const QString& id) {
+        m_id = id;
+    }
+
+    const QString& name() const {
+        return m_name;
+    }
+    void setName(const QString& name) {
+        m_name = name;
+    }
+
+    const QString& shortName() const {
+        return m_shortName;
+    }
+    void setShortName(const QString& shortName) {
+        m_shortName = shortName;
+    }
+
+    const QString& description() const {
+        return m_description;
+    }
+    void setDescription(const QString& description) {
+        m_description = description;
+    }
+
+    int index() const {
+        return m_iIndex;
+    }
+    void setIndex(int index) {
+        m_iIndex = index;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////
+    // Usage hints
+    ////////////////////////////////////////////////////////////////////////////////
+
+    const ParameterType& parameterType() const {
+        return m_parameterType;
+    }
+
+    void setParameterType(const ParameterType parameterType) {
+        m_parameterType = parameterType;
+    }
+
+    ValueScaler valueScaler() const {
+        return m_valueScaler;
+    }
+    void setValueScaler(ValueScaler valueScaler) {
```

Comment:
This appears to be the only function for this class that is not a raw setter/getter.  This class looks to me more like a struct than a full-featured class.  I think that's all right, and it would be easier to understand if it were just a bare struct.  (There's no advantage to having "private members" if they are all editable by accessors).  Issues of constness and pointers would be better handled outside the class.

Reply:
> This class looks to me more like a struct than a full-featured class. I think that's all right, and it would be easier to understand if it were just a bare struct. (There's no advantage to having "private members" if they are all editable by accessors).

I disagree, using a regular class makes it easier to extend later on if necessary. If it's a regular struct, you'll have to refactor a lot of code.